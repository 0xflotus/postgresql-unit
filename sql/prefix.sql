SELECT
  f, meter(f), kilogram(f)
FROM
  factors;

SELECT
  p, (p||'m/m')::unit AS factor,
  ('0.001 '||p||'m')::unit AS "0.001 m",
  (p||'m')::unit AS "meter",
  ('1000 '||p||'m')::unit AS "1000 m",
  ('0.001 '||p||'g')::unit AS "0.001 g",
  (p||'g')::unit AS "gram",
  ('1000 '||p||'g')::unit AS "1000 g"
FROM
  prefixes;
