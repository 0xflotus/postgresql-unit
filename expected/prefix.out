WITH
  factor(f) AS (VALUES
	(1e27), (1e24), (1e21), (1e18), (1e15), (1e12), (1e9), (1e6), (1e3),
	(1),
	(1e-3), (1e-6), (1e-9), (1e-12), (1e-15), (1e-18), (1e-21), (1e-24), (1e-27), (1e-30),
	(0))
SELECT
  f, meter(f), kilogram(f)
FROM
  factor;
                f                 |  meter  | kilogram 
----------------------------------+---------+----------
     1000000000000000000000000000 | 1e+27 m | 1e+27 kg
        1000000000000000000000000 | 1 Ym    | 1e+24 kg
           1000000000000000000000 | 1 Zm    | 1 Yg
              1000000000000000000 | 1 Em    | 1 Zg
                 1000000000000000 | 1 Pm    | 1 Eg
                    1000000000000 | 1 Tm    | 1 Pg
                       1000000000 | 1 Gm    | 1 Tg
                          1000000 | 1 Mm    | 1 Gg
                             1000 | 1 km    | 1 Mg
                                1 | 1 m     | 1 kg
                            0.001 | 1 mm    | 1 g
                         0.000001 | 1 µm    | 1 mg
                      0.000000001 | 1 nm    | 1 µg
                   0.000000000001 | 1 pm    | 1 ng
                0.000000000000001 | 1 fm    | 1 pg
             0.000000000000000001 | 1 am    | 1 fg
          0.000000000000000000001 | 1 zm    | 1 ag
       0.000000000000000000000001 | 1 ym    | 1 zg
    0.000000000000000000000000001 | 1e-27 m | 1 yg
 0.000000000000000000000000000001 | 1e-30 m | 1e-30 kg
                                0 | 0 m     | 0 kg
(21 rows)

WITH
  pfx(p) AS (VALUES
	('y'),('z'),('a'),('f'),('p'),('n'),('µ'),('mu'),('m'),('c'),('d'),
	(''),
	('da'),('h'),('k'),('M'),('G'),('T'),('P'),('E'),('Z'),('Y'))
SELECT
  p, (p||'m/m')::unit AS factor,
  ('0.001 '||p||'m')::unit AS "0.001 m",
  (p||'m')::unit AS "meter",
  ('1000 '||p||'m')::unit AS "1000 m",
  ('0.001 '||p||'g')::unit AS "0.001 g",
  (p||'g')::unit AS "gram",
  ('1000 '||p||'g')::unit AS "1000 g"
FROM
  pfx;
 p  |    factor     | 0.001 m | meter  | 1000 m  | 0.001 g  |  gram  |  1000 g  
----+---------------+---------+--------+---------+----------+--------+----------
 y  | 1e-24         | 1e-27 m | 1 ym   | 1 zm    | 1e-30 kg | 1 yg   | 1 zg
 z  | 1e-21         | 1 ym    | 1 zm   | 1 am    | 1 yg     | 1 zg   | 1 ag
 a  | 1e-18         | 1 zm    | 1 am   | 1 fm    | 1 zg     | 1 ag   | 1 fg
 f  | 1e-15         | 1 am    | 1 fm   | 1 pm    | 1 ag     | 1 fg   | 1 pg
 p  | 1e-12         | 1 fm    | 1 pm   | 1 nm    | 1 fg     | 1 pg   | 1 ng
 n  | 1e-09         | 1 pm    | 1 nm   | 1 µm    | 1 pg     | 1 ng   | 1 µg
 µ  | 1e-06         | 1 nm    | 1 µm   | 1 mm    | 1 ng     | 1 µg   | 1 mg
 mu | 1e-06         | 1 nm    | 1 µm   | 1 mm    | 1 ng     | 1 µg   | 1 mg
 m  | 0.001         | 1 µm    | 1 mm   | 1 m     | 1 µg     | 1 mg   | 1 g
 c  | 0.01          | 10 µm   | 10 mm  | 10 m    | 10 µg    | 10 mg  | 10 g
 d  | 0.1           | 100 µm  | 100 mm | 100 m   | 100 µg   | 100 mg | 100 g
    | 1             | 1 mm    | 1 m    | 1 km    | 1 mg     | 1 g    | 1 kg
 da | 10            | 10 mm   | 10 m   | 10 km   | 10 mg    | 10 g   | 10 kg
 h  | 100           | 100 mm  | 100 m  | 100 km  | 100 mg   | 100 g  | 100 kg
 k  | 1000          | 1 m     | 1 km   | 1 Mm    | 1 g      | 1 kg   | 1 Mg
 M  | 1000000       | 1 km    | 1 Mm   | 1 Gm    | 1 kg     | 1 Mg   | 1 Gg
 G  | 1000000000    | 1 Mm    | 1 Gm   | 1 Tm    | 1 Mg     | 1 Gg   | 1 Tg
 T  | 1000000000000 | 1 Gm    | 1 Tm   | 1 Pm    | 1 Gg     | 1 Tg   | 1 Pg
 P  | 1e+15         | 1 Tm    | 1 Pm   | 1 Em    | 1 Tg     | 1 Pg   | 1 Eg
 E  | 1e+18         | 1 Pm    | 1 Em   | 1 Zm    | 1 Pg     | 1 Eg   | 1 Zg
 Z  | 1e+21         | 1 Em    | 1 Zm   | 1 Ym    | 1 Eg     | 1 Zg   | 1 Yg
 Y  | 1e+24         | 1 Zm    | 1 Ym   | 1e+27 m | 1 Zg     | 1 Yg   | 1e+24 kg
(22 rows)

